crates/uv-distribution/src/workspace.rs

```diff
@@ -798,6 +782,55 @@ fn is_excluded_from_workspace(
     Ok(false)
 }
 
+/// An iterator over a [`Pattern`] in a workspace.
+///
+/// Iterates from the workspace root, returning the paths of any directories that match the pattern.
+#[derive(Debug)]
+struct WorkspaceGlobIter<'a> {
+    root: &'a Path,
+    glob: &'a Pattern,
+    iter: walkdir::IntoIter,
+}
+
+impl<'a> WorkspaceGlobIter<'a> {
+    fn new(root: &'a Path, glob: &'a Pattern) -> Self {
+        Self {
+            root,
+            glob,
+            iter: walkdir::WalkDir::new(root).into_iter(),
+        }
+    }
+}
+
+impl<'a> Iterator for WorkspaceGlobIter<'a> {
+    type Item = Result<PathBuf, walkdir::Error>;
```

Comment:
You might be able to just use `PathBuf` here? It looks like `next` below can never return an error. (This is what I would expect I think.)